src/store/tenant/index.js

```diff
@@ -176,30 +157,8 @@ const actions = {
         // already - otherwise tokens are fetched through the initial
         // authorization process
         if (getters['tenantIsSet']) {
-          let tokens
-          if (typeof window.SharedWorker !== 'undefined') {
-            tokens = await getTenantTokens(tenant.id, slug)
-
-            // Set our new auth token
-            await dispatch('auth/updateAuthorizationTokens', tokens, {
-              root: true
-            })
-          } else {
-            // Get our new auth token
-            const tenantToken = await fallbackApolloClient.mutate({
-              mutation: require('@/graphql/Tenant/tenant-token.gql'),
-              variables: {
-                tenantId: tenant.id
-              },
-              fetchPolicy: 'no-cache'
-            })
-
-            tokens = tenantToken.data.switch_tenant
-            // Set our new auth token
-            await dispatch('auth/updateAuthorization', tokens, {
-              root: true
-            })
-          }
+          const tokens = await switchTenant(tenant.id)
+          commitTokens(tokens)
```

Comment:
This is the only except where application logic accesses the auth layer because the application needs to have a way of communicating that it needs to fetch different tokens based on user interaction.